### PR TITLE
EASY-1915: describe validation messages

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-1915

#### When applied it will
* describe validation messages
* replace all occurrences of `organisation` with `organization`
* include `nonEmptyRole` in contributor validation

@DANS-KNAW/easy for review

### CODE REVIEW

File: doc/validation.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    * _error text:_ "no titles were provided"
+* **description**
+  * mandatory field
+    * _error text:_ "no description was provided"
+* **contributors**
+  * mandatory field array
+    * _error text:_ "no contributors were provided"
+  * one creator minimum
+    * _error text:_ "at least one creator is required"
+  * every nonempty contributor must have either an organization or {initial, surname}
```
Comment (jo-pol) [inline doc/validation.md]: When do you return which message? They don't reflect the rule. Same for RightsHolders.

See also 
https://github.com/DANS-KNAW/easy-deposit-api/blob/abbcc43cadb5a2e964cafb6d27e60fe1670541ec/src/test/scala/nl.knaw.dans.easy.deposit/servlets/ValidationSpec.scala#L202-L222

File: doc/validation.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    * _error text:_ "no date created was provided"
+* **audiences**
+  * mandatory field array
+    * _error text:_ "no audiences were provided"
+* **alternativeIdentifiers**
+  * no partially filled in identifiers
+    * _error text:_ "no scheme given"
+    * _error text:_ "no identifier given"
+* **relatedIdentifiers**
+  * if it has one element, only validate if either `scheme` or `value` is filled in
```
Comment (jo-pol) [inline doc/validation.md]: What if there are multiple elements? Same below.

File: doc/validation.md (hunk 1 above)
Comment (rvanheest) [reply doc/validation.md]: Hopefully f5b58f3 made this somewhat more clear. However, I don't get what you mean with the `ValidationSpec` reference.

Comment (jo-pol) [reply doc/validation.md]: With ValidationSpec I wanted to show how I validate and report certain input.

**Pull request merged**